plugins/BEdita/Core/src/Shell/Task/CheckFilesystemTask.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2017 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Shell\Task;
+
+use Cake\Console\Shell;
+
+/**
+ * Task to check permissions on filesystem.
+ *
+ * @since 4.0.0
+ */
+class CheckFilesystemTask extends Shell
+{
+
+    /**
+     * {@inheritDoc}
+     *
+     * @codeCoverageIgnore
+     */
+    public function getOptionParser()
+    {
+        $parser = parent::getOptionParser();
+        $parser
+            ->setDescription([
+                'Check filesystem permissions.',
+            ])
+            ->addArgument('paths ...', [
+                'help' => 'List of directories to check if they are writable.',
+                'required' => false,
+            ])
+            ->addOption('httpd-user', [
+                'help' => 'Manually set HTTPD user, instead of relying on automatic detection.',
+                'required' => false,
+            ]);
+
+        return $parser;
+    }
+
+    /**
+     * Perform basic checks on filesystem.
+     *
+     * @param string[] ...$paths List of paths to check if they are writable.
+     * @return bool
+     */
+    public function main(...$paths)
+    {
+        // Load paths to be checked.
+        $paths = array_unique(array_filter($paths) ?: [TMP, LOGS]);
+
+        // Detect HTTP daemon user.
+        $httpdUser = $this->getHttpdUser();
+        if (empty($httpdUser)) {
+            $this->out('=====> <warning>Unable to detect webserver user</warning>');
```

Comment:
missing coverage of empty `$httpdUser`, but we can live without :wink:

Reply:
Actually, this part is covered by `CheckFilesystemTaskTest::testExecuteAutodetectOk()` and `CheckFilesystemTaskTest::testExecuteAutodetectFail()`. However, one of this test is always skipped (if a webserver is running, fail is skipped, if a webserver is _not_ running, ok is skipped). In local environment it is easy to test both cases by running tests then turning off Apache or whatever, and re-running them again.

Reply:
Ok, I only looked at the coverage results...